File: atcoder.jp/PyPy3 (7.3.0)/abc258/abc258_d.py
from sys import stdin
input = lambda : stdin.readline().rstrip()

N, X = map(int,input().split())

ans = float('inf')
T = 0
for i in range(min(N, X)):
    A, B = map(int,input().split())
    T += A + B
    ans = min(T + (X - i - 1) * B, ans)

print(ans)

File: atcoder.jp/Nim (1.0.6)/abc258/abc258_d.nim
import strformat, strutils, sequtils, math, algorithm, tables, sets, lists, intsets, critbits, future

var
    N, T, A, B:int
    X, ans:int64
(N, X) = stdin.readLine.split.map(parseInt)

for i in 1..min(N, X):
    (A, B) = stdin.readLine.split.map(parseInt)
    T += A + B
    if i == 1:
        ans = T + (X - i) * B
    else:
        ans = min(ans, T + (X - i) * B)

echo ans